op-e2e/sgt/sgt_test.go

```diff
@@ -0,0 +1,389 @@
+		},
+		{
+			name:            "FullSGTGasPaymentWithNativeBalanceSuccess",
+			depositSgtValue: big.NewInt(10000000000000),
+			depositL2Value:  big.NewInt(10000000000000),
+			txValue:         big.NewInt(0),
+			action:          fullSGTGasPaymentWithNativeBalanceSuccess,
+		},
+		{
+			name:            "PartialSGTGasPaymentSuccess",
+			depositSgtValue: big.NewInt(1000),
+			depositL2Value:  big.NewInt(10000000000000),
+			txValue:         big.NewInt(0),
+			action:          partialSGTGasPaymentSuccess,
+		},
+		{
+			name:            "FullSGTGasPaymentAndNonZeroTxValueWithSufficientNativeBalanceSuccess",
+			depositSgtValue: big.NewInt(10000000000000),
+			depositL2Value:  big.NewInt(10000000000000),
+			txValue:         big.NewInt(10000),
+			action:          fullSGTGasPaymentAndNonZeroTxValueWithSufficientNativeBalanceSuccess,
+		},
+		{
+			name:            "PartialSGTGasPaymentAndNonZeroTxValueWithSufficientNativeBalanceSuccess",
+			depositSgtValue: big.NewInt(1000),
+			depositL2Value:  big.NewInt(10000000000000),
+			txValue:         big.NewInt(10000),
+			action:          partialSGTGasPaymentAndNonZeroTxValueWithSufficientNativeBalanceSuccess,
+		},
+		{
+			name:            "FullSGTInsufficientGasPaymentFail",
+			depositSgtValue: big.NewInt(10000),
+			depositL2Value:  big.NewInt(0),
+			txValue:         big.NewInt(0),
+			expectedErr:     errorInsufficientBalance,
+			action:          fullSGTInsufficientGasPaymentFail,
+		},
+		{
+			name:            "FullNativeInsufficientGasPaymentFail",
+			depositSgtValue: big.NewInt(0),
+			depositL2Value:  big.NewInt(10000),
+			txValue:         big.NewInt(0),
+			expectedErr:     errorInsufficientBalance,
+			action:          fullNativeInsufficientGasPaymentFail,
+		},
+		{
+			name:            "PartialSGTInsufficientGasPaymentFail",
+			depositSgtValue: big.NewInt(10000),
+			depositL2Value:  big.NewInt(10000),
+			txValue:         big.NewInt(0),
+			expectedErr:     errorInsufficientBalance,
+			action:          partialSGTInsufficientGasPaymentFail,
+		},
+		{
+			name:            "FullSGTGasPaymentAndNonZeroTxValueWithInsufficientNativeBalanceFail",
+			depositSgtValue: big.NewInt(10000000000000),
+			depositL2Value:  big.NewInt(10000),
+			txValue:         big.NewInt(10001),
+			expectedErr:     errorInsufficientBalance,
+			action:          fullSGTGasPaymentAndNonZeroTxValueWithInsufficientNativeBalanceFail,
+		},
+		{
+			name:            "PartialSGTGasPaymentAndNonZeroTxValueWithInsufficientNativeBalanceFail",
+			depositSgtValue: big.NewInt(10000),
+			depositL2Value:  big.NewInt(10000000000000),
+			txValue:         big.NewInt(10000000000000 - 10000),
+			expectedErr:     errorInsufficientBalance,
+			action:          partialSGTGasPaymentAndNonZeroTxValueWithInsufficientNativeBalanceFail,
+		},
+	}
+
+	for index, tCase := range tests {
+		t.Run(tCase.name, func(t *testing.T) {
+			tCase.action(t, ctx, int64(index), tCase.depositSgtValue, tCase.depositL2Value, tCase.txValue, sgt)
```

Comment:
What about moving the `setUpTestAccount` call into the loop, and each action continues the test with the account returned by `setUpTestAccount`?